eradiate/scenes/biosphere/_mesh_tree_element.py

```diff
@@ -0,0 +1,173 @@
+import os
+from pathlib import Path
+from typing import MutableMapping, Optional
+
+import attr
+import pint
+
+from ..core import SceneElement
+from ..spectra import Spectrum, SpectrumFactory
+from ... import validators
+from ..._attrs import documented, get_doc, parse_docs
+from ...contexts import KernelDictContext
+from ...units import unit_context_kernel as uck
+from ...units import unit_registry as ureg
+
+
+@parse_docs
+@attr.s
+class MeshTreeElement(SceneElement):
+    """
+    Container class for mesh based constituents of tree-like objects in a canopy.
+    Holds the filepath for the triangulated mesh and all parameters specifying
+    the associated BSDF.
+
+    .. important:: The triangulated mesh must be provided in .ply or .obj format.
+
+    Since mesh definition files cannot carry Pint units, the attribute ``mesh_units``
+    lets users provide the unit which their mesh is defined in. Upon kernel dict
+    creation the mesh is scaled to match the length unit used in the kernel.
+    If ``mesh_units` is not provided, no scaling of the mesh is performed.
+
+    The :meth:`MeshTreeElement.from_dict` constructor instantiates the class from
+    a configuration dictionary.
+    """
+
+    id = documented(
+        attr.ib(
+            default="mesh_tree_element",
+            validator=attr.validators.optional(attr.validators.instance_of(str)),
+        ),
+        doc=get_doc(SceneElement, "id", "doc"),
+        type=get_doc(SceneElement, "id", "type"),
+        default="'mesh_tree_element'",
+    )
+
+    mesh_filename = documented(
+        attr.ib(
+            default=None,
+            converter=attr.converters.optional(Path),
+            validator=validators.path_exists,
+        ),
+        doc="Path to the triangulated mesh data file.",
+        type="path-like",
+        default="None",
+    )
+
+    @mesh_filename.validator
+    def _mesh_filename_validator(self, attribute, value):
+        if not value.suffix in [".obj", ".ply"]:
+            raise ValueError(
+                f"While validating {attribute.name}: File extension must be '.obj'"
+                f"or '.ply', got {value.suffix}"
+            )
```

Comment:
For the "neat" validation update:
```suggestion
    def _mesh_filename_validator(self, attribute, value):
        if value is None:
            raise ValueError("'mesh_filename' is required")
        
        validators.path_exists(self, attribute, value)       

        if not value.suffix in [".obj", ".ply"]:
            raise ValueError(
                f"While validating {attribute.name}: File extension must be '.obj'"
                f"or '.ply', got {value.suffix}"
            )
```